tests/integration/vendorCompanies.test.js

```diff
@@ -62,3 +63,98 @@ describe('VENDOR COMPANY ROUTES - GET /vendorcompany', () => {
     });
   });
 });
+
+describe('VENDOR COMPANY ROUTES - PUT /vendorcompanies', () => {
+  let authToken;
+  beforeEach(populateDB);
+
+  describe('TRAINING_ORGANISATION_MANAGER', () => {
+    beforeEach(async () => {
+      authToken = await getToken('training_organisation_manager');
+    });
+
+    const payloads = [
+      { key: 'name', value: 'Campanil' },
+      {
+        key: 'address',
+        value: {
+          fullAddress: '12 rue de ponthieu 75008 Paris',
+          zipCode: '75008',
+          city: 'Paris',
+          street: '12 rue de Ponthieu',
+          location: { type: 'Point', coordinates: [2.377133, 48.801389] },
+        },
+      },
+      { key: 'siret', value: '12345678901235' },
+    ];
+    payloads.forEach((payload) => {
+      it(`should update vendor company ${payload.key}`, async () => {
+        const response = await app.inject({
+          method: 'PUT',
+          url: '/vendorcompanies',
+          payload: { [payload.key]: payload.value },
+          headers: { Cookie: `alenvi_token=${authToken}` },
+        });
+
+        expect(response.statusCode).toBe(200);
+
+        const vendorCompany = await VendorCompany.countDocuments({ [payload.key]: payload.value });
+        expect(vendorCompany).toBeTruthy();
+      });
+    });
+
+    const wrongValues = [
+      { key: 'name', value: '' },
+      {
+        key: 'address',
+        value: {
+          fullAddress: '',
+          zipCode: '75008',
+          city: 'Paris',
+          street: '12 rue de Ponthieu',
+          location: { type: 'Point', coordinates: [2.377133, 48.801389] },
+        },
+      },
+      {
+        key: 'address',
+        value: '12 rue de ponthieu 75008 Paris',
+      },
+      { key: 'siret', value: '13244' },
+    ];
+    wrongValues.forEach((payload) => {
+      it('should update vendor company infos', async () => {
```

Comment:
Should not update vendor company `{key}` with wrong value ? 

On pourrait eventuellement mettre la valeur mais pour l'adresse ca me parait complexe 🤔

Reply:
done